sklearn/utils/validation.py

```diff
@@ -1140,14 +1140,19 @@ def _check_y(y, multi_output=False, y_numeric=False, estimator=None):
     return y
 
 
-def column_or_1d(y, *, warn=False):
+def column_or_1d(y, *, dtype=None, warn=False):
     """Ravel column or 1d numpy array, else raises an error.
 
     Parameters
     ----------
     y : array-like
        Input data.
 
+    dtype : data-type, default=None
+        Data type for `y`.
```

Comment:
What is the behavior of `column_or_1d` when dtype=None? Is it this one?
```suggestion
    dtype : data-type, default=None
        Data type for `y`.
        When dtype is None, dtype is inferred from the elements of `y`.
```

Reply:
Yea, it is inferred from the elements of `y`:

XREF: [np.asarray docs](https://numpy.org/doc/stable/reference/generated/numpy.asarray.html?highlight=asarray#numpy.asarray)